### PR TITLE
Issue 14762 - Reduce confusion for libcurl after HTTP request.

### PR DESCRIPTION
**Pull request by karronoli**
After http request is done, Clear curl option for below problems.

- Prefer CUSTOM request by libcurl
https://github.com/bagder/curl/blob/master/lib/http.c#L1834
- Change any method to PUT after PUT by libcurl
https://github.com/bagder/curl/blob/master/lib/http.c#L1830

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
LGTM

File: std/net/curl.d [hunk 1]
```diff
@@ -2261,32 +2261,46 @@ struct HTTP
     {
         p.status.reset();
 
+        CurlOption opt;
+        scope (exit)
```
Comment (DmitryOlshansky) [inline std/net/curl.d]: So opt is set on exit... yet the `opt` variable is changed later on. The control flow is confusing at best maybe just put p.curl.clear(opt) after the switch? Or do p.curl.xyz throw (I guess not - this is C library)?

Comment (karronoli) [inline std/net/curl.d]: Yes. p.curl.perform may throw. but I think that this scope(exit) statement is not so good too...

File: std/net/curl.d [hunk 2]
```diff
@@ -2261,32 +2261,46 @@ struct HTTP
...
             break;
         case Method.trace:
             p.curl.set(CurlOption.customrequest, "TRACE");
+            opt = CurlOption.customrequest;
             break;
         case Method.connect:
             p.curl.set(CurlOption.customrequest, "CONNECT");
+            opt = CurlOption.customrequest;
             break;
         }
```
Comment (karronoli) [inline std/net/curl.d]: I will move the scope (exit) statement here. Do you think the control flow is improved?

**Comment by karronoli** [comment]
I will move the scope (exit) statement after the switch. Do you think the control flow is improved?

**Comment by DmitryOlshansky** [comment]
> I will move the scope (exit) statement after the switch. Do you think the control flow is improved?
> f72ec92

This makes it much better.

**Comment by karronoli** [comment]
Thank you so much for advice! I've squashed the commits.

**Comment by DmitryOlshansky** [comment]
@karronoli are you on windows? Could you also take a look at 
https://issues.dlang.org/show_bug.cgi?id=14793

And since you are working on std.net.curl, it would be great to try and fix this nasty thing:
https://issues.dlang.org/show_bug.cgi?id=13573

**Comment by karronoli** [comment]
OK. I also help something.

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**